core-primitives/node-api-extensions/src/metadata.rs

```diff
@@ -0,0 +1,190 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::error::{Error, Result};
+use codec::{Decode, Encode};
+use sp_core::storage::StorageKey;
+use substrate_api_client::{Metadata, MetadataError};
+
+/// Handle SGX compatible substrate chain metadata.
+
+///Pallet's names
+pub static TEEREX: &str = "Teerex";
+pub static SIDECHAIN: &str = "Sidechain";
+pub static SYSTEM: &str = "Sidechain";
+
+#[derive(Encode, Decode, Debug, Clone)]
+pub struct NodeMetadata {
+	node_metadata: Option<Metadata>,
+	runtime_spec_version: u32,
+	runtime_transaction_version: u32,
+}
+
+impl Default for NodeMetadata {
+	fn default() -> Self {
+		Self { node_metadata: None, runtime_spec_version: 0, runtime_transaction_version: 0 }
+	}
+}
+
+impl NodeMetadata {
+	pub fn new(
+		node_metadata: Metadata,
+		runtime_spec_version: u32,
+		runtime_transaction_version: u32,
+	) -> Self {
+		Self {
+			node_metadata: Some(node_metadata),
+			runtime_spec_version,
+			runtime_transaction_version,
+		}
+	}
+
+	///Teerex storages
+	pub fn enclave_count_storage_key(&self) -> Result<StorageKey> {
+		self.storage_value_key(TEEREX, "EnclaveCount")
+	}
+	pub fn enclave_registry_storage_map_key(&self, index: u64) -> Result<StorageKey> {
+		self.storage_map_key(TEEREX, "EnclaveRegistry", &index)
+	}
+
+	///System storages
+	pub fn system_account_storage_key(&self) -> Result<StorageKey> {
+		self.storage_value_key(SYSTEM, "Account")
+	}
+	pub fn system_account_storage_map_key(&self, index: u64) -> Result<StorageKey> {
+		self.storage_map_key(SYSTEM, "Account", &index)
+	}
+
+	/// Get the storage keys corresponding to a storage over the metadata
+	/// Storage value key
+	pub fn storage_value_key(
+		&self,
+		storage_prefix: &'static str,
+		storage_key_name: &'static str,
+	) -> Result<StorageKey> {
+		match &self.node_metadata {
+			None => return Err(Error::MetadataNotSet),
+			Some(m) => m
+				.storage_value_key(storage_prefix, storage_key_name)
+				.map_err(|e| Error::NodeMetadata(e)),
+		}
+	}
+	/// Storage map key
```

Comment:
Comments like these are redundant, they do not add any information to the function (since the function name says it already)